conans/test/integration/graph/core/test_build_requires.py

```diff
@@ -405,6 +405,24 @@ def test_trait_aggregated(self):
         _check_transitive(lib, [(gtest, True, True, False, False),
                                 (zlib, True, True, False, False, False)])
 
+    def test_test_require_loop(self):
+        # https://github.com/conan-io/conan/issues/15412
+        self._cache_recipe("gtest/1.11", GenConanfile())
+        self._cache_recipe("abseil/1.0", GenConanfile().with_test_requires("gtest/[>=1 <1.14]"))
+        self._cache_recipe("gtest/1.14", GenConanfile().with_requires("abseil/1.0"))
+        deps_graph = self.build_graph(GenConanfile("opencv", "1.0").with_require("gtest/1.14"))
```

Comment:
maybe you wanted `with_test_require("gtest/1.14")` here?

Reply:
Sure, I was testing that the issues and fixes were the same for requires and test_requires and I left this without restoring its original ``test_requires``. Test fixed, thanks for noticing!